src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -66,19 +71,29 @@ internal sealed partial class VisualStudioMetadataReferenceManager : IWorkspaceS
     /// </summary>
     private IVsSmartOpenScope? SmartOpenScopeServiceOpt { get; set; }
 
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
     public VisualStudioMetadataReferenceManager(
         IServiceProvider serviceProvider,
-        TemporaryStorageService temporaryStorageService)
+        IThreadingContext threadingContext,
+        VisualStudioWorkspace workspace)
     {
         _runtimeDirectories = GetRuntimeDirectories();
 
-        _xmlMemberIndexService = (IVsXMLMemberIndexService)serviceProvider.GetService(typeof(SVsXMLMemberIndexService));
-        Assumes.Present(_xmlMemberIndexService);
+        _xmlMemberIndexService = new VSThreading.AsyncLazy<IVsXMLMemberIndexService>(
```

Comment:
Rather than having to write this explicitly, you can use `IVsService<>` as a magic MEF import that will do the actual GetService for you.

Reply:
Doesn't IVsService only have async getters? This needs to be exposed as in a sync manner.